File: cat2/balans/python/fvogels.py
def pairs(xs):
    for i in range(len(xs)):
        for j in range(i + 1, len(xs)):
            yield (xs[i], xs[j])

def solve(goal, weights):
    return any(x + y == goal for x, y in pairs(weights))


n = int(input())

for index in range(n):
    input()
    weights = [ int(x) for x in input().split(' ') ]
    goal = int(input())
    
    if solve(goal, weights):
        result = "JA"
    else:
        result = "NEEN"

    print("{} {}".format(goal, result))

File: cat2/balans/ruby/fvogels.rb
def solve(ns, goal)
  ns.sort!

  i = 0
  j = ns.length - 1

  while i < j
    case
    when ns[i] + ns[j] < goal
    then i += 1

    when ns[i] + ns[j] > goal
    then j -= 1

    else return 'JA'
    end
  end
  
  'NEEN'
end

(1..gets.to_i).each do |index|
  gets
  weights = gets.split(/ /).map(&:to_i)
  goal = gets.to_i

  puts "#{goal} #{solve(weights, goal)}"
end
